yarn-project/simulator/src/public/avm/avm_simulator.ts

```diff
@@ -66,8 +66,8 @@ export class AvmSimulator {
 
   // Factory to have a proper function name in the logger. Retrieving the name is asynchronous and
   // cannot be done as part of the constructor.
-  public static async build(context: AvmContext): Promise<AvmSimulator> {
-    const simulator = new AvmSimulator(context);
+  public static async build(context: AvmContext, enableTallying: boolean = false): Promise<AvmSimulator> {
+    const simulator = new AvmSimulator(context, enableTallying);
```

Comment:
I don't like it very much that tallying/metrics got so tied with the core simulator (even before this PR). Ideally I think we'd have something a bit more like the merkle DBs, you can build one that does hinting, and one that does not. We could (in theory) have a simulator that collects metrics and forwards. IDK if it's as easy if we need info about each opcode etc though, but we could consider at some point.